include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -645,9 +646,11 @@ def JoinOp : Handshake_Op<"join", [
 def I4 : I<4>;
 def I4Attr : SignlessIntegerAttrBase<I4, "4-bit integer attribute">;
 
+def HasOneOperand : Constraint<CPred<"$_self.size() == 1">, "has one operand">;
 def HasOneResult : Constraint<CPred<"$_self.size() == 1">, "has one result">;
 
-def EliminateSimpleMergesPattern : Pat<(MergeOp $a), (replaceWithValue $a)>;
+def EliminateSimpleMergesPattern : Pat<(MergeOp $a), (replaceWithValue $a),
+                                       [(HasOneOperand:$a)]>;
```

Comment:
I didn't intend to change the Merge canonicalization, but after writing the test case, I found that the resulting Merge was going to be replaced by this canonicalization pattern even though it had more than one input, which led to an assertion and crash.

Reply:
I think this should be the right pattern. The previous pattern worked because StandardToHandshake only generates single input MergOp...